src/development/ui/advanced/shaders.md

```diff
@@ -0,0 +1,228 @@
+Flutter applications and packages via the FragmentProgram API. These can be
+used to provide rich graphical effects beyond those provided by the Flutter SDK.
+
+## Adding Shaders to an Application
+
+Shaders must be declared in the new `shaders` section of your application or
+package pubspec.yaml file. This allows the Flutter command line tool to compile
+the shader to the appropriate backend format, as well as generate the necessary
+runtime metadata. When running in debug mode, changes to a shader program will
+trigger a recompilation and update of the shader during hot reload or hot
+restart.
+
+```yaml
+flutter:
+  shaders:
+    - shaders/myshader.frag
+```
+
+Shaders from packages can be included using the same technique as other assets.
+
+#### Loading shaders at runtime
+
+FragmentPrograms can be loaded at runtime using the [FragmentProgram.fromAsset]
+constructor. The key should be the file path specified in the pubspec.yaml
+above.
+
+```dart
+void loadMyShader() async {
+  var program = await FragmentProgram.fromAsset('shaders/myshader.frag');
+}
+```
+
+The FragmentProgram object can be used to create one or more FragmentShader
+instances. A FragmentShader represents a particular combination of a
+program and bound uniforms.
+
+```dart
+void updateShader(Canvas canvas, Rect rect, FragmentProgram program) {
+  var shader = program.createShader();
+  shader.setFloat(0, 42.0);
+  canvas.drawRect(rect, Paint()..shader = shader);
+}
+```
+
+## Canvas API
+
+Fragment shaders can be used with most Canvas APIs by setting [Paint.shader].
+For example, when using [Canvas.drawRect] the shader will be evaluated for all
+fragments within the rectangle. For an API like [Canvas.drawPath] with a
+stroked path, the shader will be evaluated for all fragments within the stroked
+line. Some APIs such as [Canvas.drawImage] ignore the value of the shader.
+
+```dart
+void paint(Canvas canvas, Size size, FragmentShader shader) {
+  // Draws a rectangle with the shader used as a color source.
+  canvas.drawRect(
+    Rect.fromLTWH(0, 0, size.width, size.height),
+    Paint()..shader = shader,
+  );
+
+  // Draws a stroked rectangle with the shader only applied to the fragments
+  // that lie within the stroke.
+  canvas.drawRect(
+    Rect.fromLTWH(0, 0, size.width, size.height),
+    Paint()
+      ..style = PaintingStyle.stroke
+      ..shader = shader,
+  )
+}
+
+```
+
+## Authoring Shaders
+
+Shaders are authored as GLSL source files. Any version from 460 down to 100
+is supported, though some available features are restricted. For the rest
+of the examples in this document `460 core` will be used.
+
+* UBOs and SSBOs are unsupported.
+* `sampler2D` is the only supported sampler type
+* Only the two argument version of `texture` (sampler and uv) is supported.
+* No varyings may be declared.
+  * Note: A position varying is available for each shader
+* All precision hints are ignored when targeting Skia.
+* Unsigned integers and booleans are unsupported.
```

Comment:
I took most of these suggestions except that I changed the line on varyings. I don't think I need to call out the position varying since I talk about getting the current position below